aviatrix/resource_aviatrix_dcf_trustbundle.go

```
@@ -0,0 +1,125 @@
+		DeleteWithoutTimeout: resourceAviatrixDCFTrustBundleDelete,
+
+		Schema: map[string]*schema.Schema{
+			"display_name": {
+				Type:         schema.TypeString,
+				Required:     true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				Description:  "Display name for the DCF trust bundle.",
+			},
+			"bundle_content": {
+				Type:         schema.TypeString,
+				Required:     true,
+				Sensitive:    true,
+				ValidateFunc: validation.StringIsNotWhiteSpace,
+				Description:  "The CA bundle content in PEM format.",
+			},
+			"bundle_id": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "The unique identifier for the trust bundle.",
+			},
+			"created_at": {
+				Type:        schema.TypeString,
+				Computed:    true,
+				Description: "ISO 8601 timestamp when the trust bundle was created.",
+			},
+		},
+	}
+}
+
+func marshalDCFTrustBundleInput(d *schema.ResourceData) *goaviatrix.TrustBundleItemRequest {
+	return &goaviatrix.TrustBundleItemRequest{
+		DisplayName:   d.Get("display_name").(string),
+		BundleContent: d.Get("bundle_content").(string),
+	}
+}
+
+func resourceAviatrixDCFTrustBundleCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	trustBundleRequest := marshalDCFTrustBundleInput(d)
+
+	bundleID, err := client.CreateDCFTrustBundle(ctx, trustBundleRequest)
+	if err != nil {
+		return diag.Errorf("failed to create DCF Trust Bundle: %s", err)
+	}
+
+	d.SetId(bundleID)
+
+	// Call read to populate computed fields
+	return resourceAviatrixDCFTrustBundleRead(ctx, d, meta)
+}
+
+func resourceAviatrixDCFTrustBundleRead(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	client := meta.(*goaviatrix.Client)
+
+	bundleID := d.Id()
+
+	trustBundle, err := client.GetDCFTrustBundle(ctx, bundleID)
+	if err != nil {
+		if errors.Is(err, goaviatrix.ErrNotFound) {
```

Why is it the right behavior to not error if we don't find the bundle? If the terraform state is current, then it should be there. Silently returning nil is a worry.

---

Updated to return error

---

If we return error on both paths why do we need special handling here?

---

I think we talked about this above, if the error is not found or not existing for a trust bundle we should still return 404 which is ErrNotFound, if we return ErrNotFound the unit test passes well, as TF expects this type when checking if the resource exists or not after destroy.